services/agent/tests/conftest.py

```diff
@@ -182,14 +182,14 @@ def caplog_info_debug(caplog: LogCaptureFixture) -> Iterable[LogCaptureFixture]:
         yield caplog
 
 
-@pytest.fixture
+@pytest.fixture(scope="module")
 def mocked_s3_server_url() -> Iterator[HttpUrl]:
     """
     For download links, the in-memory moto.mock_s3() does not suffice since
     we need an http entrypoint
     """
     # http://docs.getmoto.org/en/latest/docs/server_mode.html#start-within-python
-    server = ThreadedMotoServer(ip_address=get_localhost_ip(), port=9000)
+    server = ThreadedMotoServer(ip_address=get_localhost_ip(), port=9339)
```

Comment:
have you considered  ``unused_port`` fixture? 
This comes with aiohttp but if you do not have that library, see ``packages/service-library/tests/fastapi/conftest.py``